pkg/metadata/scheduler.go

```diff
@@ -26,6 +26,7 @@ type Scheduler struct {
 	hostname      string
 	tickers       []*time.Ticker
 	healthHandles []*health.Handle
+	stopChannels  []chan struct{}
```

Comment:
my two cents: Scheduler should have store "parent" `context.Context` and its cancel function;  created with `ctx, cancelFunc := context.WithCancel(context.Background())` and then for each added Collector, create a child context from the parent context.

Like this the cascading "cancel|stop" is handle for you when you call the cancel function of the parent context.

Reply:
Ok will look at that.